### ISSUE TITLE
Detailed command to run RO with PreRes-18

### ISSUE THREAD

**Issue by VertexC** [cvlab-columbia/SelfSupDefense#3]
I used RO public ckpt [cifar10_l2_eps128.pth](https://drive.google.com/drive/folders/110JHo_yH9zwIf1b12jKoG6dRonrow9eA) and train preResnet18SSL model with `env CUDA_VISIBLE_DEVICES=0,1,2,3 python3 cifar10_defense.py --md_path=checkpoint/cifar10_l2_eps128.pth    --batch-size=1024 --contrastive_bs=1024 --epochs=100 --res18`, but I failed to get similar results shown in the paper (28.45 vs 30.99 in my case, 52.40 vs 54.59 in paper).

Is there any hyperparameter I need to change here? Thx!

**Comment by ChengzhiCU** [comment]
I believe you use the wrong normalization. For input image's normalization
function, do you subtract mean and divided by var? Undo yours should give
you the right results.

**Comment by VertexC** [comment]
I used the normalization 
```
def normalize(X):
    return transforms.functional.normalize(X, (0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010))
```
from which I can get 90% acc without attack from RO's ckpt I downloaded, otherwise the acc was 30%. Or did you mean I should retrain RO by without normalization? Thx!

**Comment by ChengzhiCU** [comment]
One of them should work. 28.45 does not reproduce RO's results.So you
should use the other normalization.

**Comment by VertexC** [comment]
Thx for the reply. But I am still confused, which normalization should I use? I just checked their cifar training code, it seems that they are using `(X - mu)/std`.

**Comment by ChengzhiCU** [comment]
I am wondering if you solved this. You should use the normalization from https://github.com/locuslab/robust_overfitting/blob/master/train_cifar.py, first you need to reproduce their number, then you can apply our approach for better robustness.

**Comment by ChengzhiCU** [comment]
Yes, I also checked my code and it uses (X - mu)/std. It seems your low acc
is due to not reproducing the robustoverfit model in the first place. Try
run their code and see if you can reproduce it, then apply our method on
top.
